erpnext/stock/doctype/item/item.py

```diff
@@ -806,6 +807,78 @@ def validate_has_variants(self):
 			if frappe.db.exists("Item", {"variant_of": self.name}):
 				frappe.throw(_("Item has variants."))
 
+	def validate_attributes_in_variants(self):
+		if not self.has_variants:
```

Comment:
```suggestion
		if not self.has_variants or self.get("__islocal"):
```
unsaved items won't have old doc.attributes